### PR TITLE
Introduce builtin_shared and use it to fix #4778

### LINKED ISSUE #4778: Crash on latest git due to recent commit.
> ServerError: Lua: Runtime error from mod '' in callback item_OnPlace(): /usr/local/share/minetest/builtin/game/item.lua:278: attempt to call global 'check_attached_node' (a nil value)

A recent commit made this function and a few others local to their own file, thus hiding them from the rest of the engine and games.

Edit: commit is this one: 649448a2a91fbf3e944b2f2e739f4e2292af1df0

### PR DESCRIPTION
**Pull request by est31**

### CODE REVIEW

File: builtin/game/falling.lua [hunk 1]
```diff
@@ -1,5 +1,7 @@
 -- Minetest: builtin/item.lua
 
+local builtin_shared = ...
```
Comment (paramat) [inline builtin/game/falling.lua]: Interesting, why are these lines needed? No objection to these though.

Comment (rubenwardy) [inline builtin/game/falling.lua]: the alternative would be to delete a global variable or member of a global table

Comment (paramat) [inline builtin/game/falling.lua]: Ok.

**Comment by paramat** [comment]
Addresses #4778

**Comment by paramat** [comment]
A request: please can the bug fixed be explained in words in the commit message, not just an issue number?

**Comment by paramat** [comment]
Will test.

**Comment by paramat** [comment]
I can now place nodes :+1:

**Comment by est31** [comment]
afc48c802a81b95b11aef3ad0141a56f6a184979
